### PR TITLE
fix: parsing of large files > 512 MB [ZEND-3117]

### PR DESCRIPTION
**Pull request by marcolink**
Parsing large input files (> 512 MB) caused `JSON.parse` to throw an error:
```
Error: Cannot create a string longer than 0x1fffffe8 characters
```
this is a node internal hard limit for parsing strings to objects. 

This PR introduces a streaming library to allow bigger input files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -196,3 +196,6 @@ $RECYCLE.BIN/
 
 # ignore left over error files
 errorlogfile.json
+
+# test input
+large-space.json
```
Comment (andreascful) [inline .gitignore:201]: Should we really have this permanently in the gitignore? :)

File: lib/parseOptions.js [hunk 2]
```diff
@@ -74,10 +75,12 @@ export default function parseOptions (params) {
   } else {
     options.errorLogFile = resolve(process.cwd(), options.errorLogFile)
   }
-
-  // Further processing
   options.accessToken = options.managementToken
-  options.content = options.content || JSON.parse(fs.readFileSync(options.contentFile))
+
+  if (!options.content) {
```
Comment (andreascful) [inline lib/parseOptions.js:80]: Should we maybe add a comment here briefly explaining why it was done the way it was done? Just thinking in terms of someone looking at this in two years and wondering why it's a stream.

**Review by andreascful — APPROVED** [review]
Minor nitpicks. Looks great though. Enables processing of large imports! :)

File: .gitignore (hunk 1 above)
Comment (marcolink) [reply .gitignore]: anyway - deleted it!

**Pull request merged**